src/file_handler.rs

```diff
@@ -123,22 +124,94 @@ fn config_dir_path() -> String {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use file_handler::ConfigManagement;
+    use file_handler::FileManagement;
     use file_handler::FileSystem;
+    use std::fs;
     use std::io;
 
     struct MockFileSystem;
+    struct MockFileHandler;
 
     impl FileSystem for MockFileSystem {
         fn create_dir(&self, _path: &str) -> io::Result<()> {
             Ok(())
         }
     }
 
+    impl ConfigManagement for MockFileHandler {
+        fn config_dir_create(&self) -> io::Result<String> {
+            Ok(config_dir_path())
+        }
+
+        fn config_dir_exists(&self) -> bool {
+            self.file_exists(&config_dir_path())
+        }
+
+        fn config_read(&self, _file: ConfigFile) -> io::Result<String> {
+            let contents = String::from("~/idea");
+            Ok(contents)
+        }
+
+        fn config_write(&self, _file: ConfigFile, _value: String) -> io::Result<()> {
+            Ok(())
+        }
+    }
+
+    impl FileManagement for MockFileHandler {
+        fn file_exists(&self, path: &str) -> bool {
+            fs::metadata(path).is_ok()
+        }
+
+        fn file_rm(&self, _file: ConfigFile) -> io::Result<()> {
+            Ok(())
+        }
+    }
+    // test for FileSystem methods
     #[test]
-    fn it_works() {
+    fn create_dir() {
         let _fs = MockFileSystem {};
-        let actual = _fs.create_dir("irrelevant");
+        let actual = _fs.create_dir("./test");
```

Comment:
The reason I chose the string `irrelevant` here is because then when I read the test I know that the value of the string is not relevant to the test passing. It's just a small thing I like to do.

This might be a poor example but when you have lots of mocked data in a test it's nice to see what data is relevant for the test to pass, and what's irrelevant. No need to change it though, just explaining my thought process.